sdk/dotnet/Pulumi/Serialization/Deserializer.cs

```diff
@@ -217,14 +217,12 @@ private static Asset DeserializeAsset(Value value)
                 return false;
             }
 
-            string? urn;
-            if (!TryGetStringValue(value.StructValue.Fields, Constants.ResourceUrnName, out urn))
+            if (!TryGetStringValue(value.StructValue.Fields, Constants.ResourceUrnName, out var urn))
             {
                 throw new InvalidOperationException("Value was marked as a Resource, but did not conform to required shape.");
             }
 
-            string? version;
-            if (!TryGetStringValue(value.StructValue.Fields, Constants.ResourceVersionName, out version)) {
+            if (!TryGetStringValue(value.StructValue.Fields, Constants.ResourceVersionName, out var version)) {
```

Comment:
Awesome cleanup across the files. Do you mind doing this in a separate PR? Thanks a ton!

Reply:
Will do. Removed.